pex/resolver.py

```diff
@@ -142,31 +142,48 @@ def map_packages(resolved_packages):
 
 
 class Resolver(object):
-  """Interface for resolving resolvable entities into python packages."""
+  """Interface for resolving resolvable entities into python packages.
+
+  :param versions: a list of string versions, of the form ["33", "32"],
+    or None. The first version will be assumed to support our ABI.
+  :param platform: specify the exact platform you want valid
+    tags for, or None. If None, use the local system platform.
+  :param impl: specify the exact implementation you want valid
+    tags for, or None. If None, use the local interpreter impl.
+  :param abi: specify the exact abi you want valid
+    tags for, or None. If None, use the local interpreter abi.
+  """
 
   class Error(Exception): pass
 
-  @classmethod
-  def filter_packages_by_interpreter(cls, packages, interpreter, platform):
+  def filter_packages_by_interpreter(self, packages):
     return [package for package in packages
-        if package.compatible(interpreter.identity, platform)]
+            if package.compatible(self._supported_tags)]
 
-  def __init__(self, allow_prereleases=None, interpreter=None, platform=None):
+  def __init__(self, allow_prereleases=None, interpreter=None, versions=None,
+               platform=None, impl=None, abi=None):
     self._interpreter = interpreter or PythonInterpreter.get()
-    self._platform = platform or Platform.current()
+    self._platform = platform
     self._allow_prereleases = allow_prereleases
+    self._impl = impl
+    self._abi = abi
+    self._versions = versions
```

Comment:
It's just because of `get_supported`. I believe there was a point where I had myself convinced we would need to pass multiple versions to get prior minor versions in the list, but I don't think we actually need it. I'll get rid of the plural everywhere but `get_supported` (since that's copied pip code).